### ISSUE TITLE
BUG: Names of all my proteins in tsv file are "."

### ISSUE THREAD

**Issue by thdurand4** [ccdmb/predector#82]
Hello, 

Thanks for your tools it did well for me except one thing, when i go in my repository results and i open all my tsv files , the "name" column is all time "." and not the name of my proteins. 

I got the name of my proteins only in the gff3 files.  

It's an example of what i have. 

![image](https://user-images.githubusercontent.com/91953714/166420631-542ab6b7-77ee-4204-9479-bc333a7544ae.png)
 
Thanks you for your answer.

**Comment by darcyabjones** [comment]
Hi @thdurand4,

Thanks for filing the issue.
I knew about this (see #78) and have a [fix](https://github.com/ccdmb/predector-utils/commit/43ada58414cc794324a696a533e2ca58c1dd02fb), but unfortunately haven't made the time to update the pipeline.
I think I'm definitely past the critical point though, so I'll try to fix it this afternoon and tomorrow.

If you're comfortable installing python packages yourself, I can take you through getting the tables from the ldjson file if you like?
That would mean you don't have to run the pipeline again.

**Comment by thdurand4** [comment]
Thanks for your answer @darcyabjones. 

Sorry for that i did not see #78.  Thanks you so much for that. 

I can try, which packages i need to install ? 

But if can't install packages when you will fix it , do i need to install again the pipeline ? or i just need to change version with nextflow with the arg -r ?

**Comment by darcyabjones** [comment]
No apologies necessary,
It wasn't a particularly descriptive title, and it was nearly a month ago now.

Just FYI because it wasn't clear from your question if you knew, the [`-ranked.tsv` table is the main output of the pipeline](https://github.com/ccdmb/predector/wiki#-rankedtsv) where all of the results are aggregated and it should have the names in there.

If you wanted to re-run the pipeline unfortunately you'd have to re-build all of the dependencies.
It's one of the quirks that I haven't been able to work around yet because of licensing and complex dependencies.

So to decode the tables you need to install the updated `predectorutils` package (https://pypi.org/project/predectorutils/, https://github.com/ccdmb/predector-utils, https://anaconda.org/predector/predectorutils) 

This is how I'd do it with plain python in a [virtual environment](https://virtualenv.pypa.io/en/latest/), but you could install it with conda too.

```
mkdir temporary_predector_fix
cd temporary_predector_fix

python3 -m venv env
source env/bin/activate
python3 -m pip install "predectorutils==0.8.3"
```

For each `.ldjson` result file (which i assume here is called `test_set.ldjson`), you'll need to load it into an SQLite database.
Then you can get fresh new tables with the names from that SQLite database.

```bash
# Depending on how large your proteome is, this might take a while.
predutils load_db test_set.db ./test_set.ldjson
predutils tables -t "test_set/test_set-{analysis}.tsv" test_set.db
```

So the new tables will all be in the directory "test_set".
We set the template for the filenames to the `-t` flag, if you change it make sure you keep the `{analysis}` bit, otherwise it'll just overwrite the same file.
Anything in here called `test_set` is safe to change to be specific for your data.
A bash for loop should be fine if you've got lots of proteomes to deal with.


When you're done and have copied what you need to somewhere more permanent, you can just delete the whole folder to clean up (e.g `cd .. && rm -rf -- temporary_predector_fix`).

Hopefully that works for you, but happy to help if you have trouble.
I'll try to update the pipeline tomorrow, but I'm anticipating some trouble updating dependencies to a new python version.

Ta,
Darcy

**Comment by thdurand4** [comment]
thank you for the explanations, 

I have just one problem when i try this command : predutils load_db test_set.db ./test_set.ldjson on my .ldjson file i got an error : 

![image](https://user-images.githubusercontent.com/91953714/166458585-eac80ffd-64f6-4c95-9ee5-dc39665f91c0.png)

**Comment by darcyabjones** [comment]
Oh sorry about that.
Looks like either my syntax is out of date or you've got a very old version of SQLite.
What version of SQlite do you have installed?
Try `sqlite3 -version`.

If there is a newer version of SQLite3 available via a module see loading that resolves the issue.
I'll see if I can track it down though.

**Comment by thdurand4** [comment]
I updated my sqlite3 versions and it's work fine for me thanks you so much for your help

**Comment by darcyabjones** [comment]
Great!
Sorry for the trouble.

I'm working to push a new release ASAP so you shouldn't have to do this in the future.
I'll leave this issue open until I do so that anyone else with the same problem can find it easily.

All the best,
Darcy

**Comment by darcyabjones** [comment]
Thanks again for reporting this!
For next time this should now be fixed in #83
